packages/toolkit/src/query/core/buildMiddleware/windowEventHandling.ts

```diff
@@ -35,24 +39,27 @@ export const build: SubMiddlewareBuilder = ({
         const querySubState = queries[queryCacheKey]
         const subscriptionSubState = subscriptions[queryCacheKey]
 
-        if (
-          !subscriptionSubState ||
-          !querySubState ||
-          querySubState.status === QueryStatus.uninitialized
-        )
-          return
-
-        const shouldRefetch =
-          Object.values(subscriptionSubState).some(
-            (sub) => sub[type] === true
-          ) ||
-          (Object.values(subscriptionSubState).every(
-            (sub) => sub[type] === undefined
-          ) &&
-            state.config[type])
-
-        if (shouldRefetch) {
-          api.dispatch(refetchQuery(querySubState, queryCacheKey))
+        if (!subscriptionSubState || !querySubState) return
```

Comment:
On re-reading this... shouldn't this `return` always have been a `continue`?

Reply:
Very good catch, I'm surprised this hasn't caused more severe issues. I'll adjust it soon.